app/views/project/editor/history/fileTreeV2.pug

```diff
@@ -56,7 +56,6 @@ script(type="text/ng-template", id="historyFileEntityTpl")
 				file-entity="childEntity"
 			)
 
-script(type="template", id="file_action_edited_str") #{ translate('file_action_edited') }
-script(type="template", id="file_action_renamed_str") #{ translate('file_action_renamed') }
-script(type="template", id="file_action_created_str") #{ translate('file_action_created') }
-script(type="template", id="file_action_deleted_str") #{ translate('file_action_deleted') }
+- var fileActionI18n = ['edited', 'renamed', 'created', 'deleted'].reduce((acc, i) => {acc[i] = translate('file_action_' + i); return acc}, {})
```

Comment:
I think a `map` would be more appropriate here? Something like:

```suggestion
- var fileActionI18n = ['edited', 'renamed', 'created', 'deleted'].map((action) => translate('file_action_' + action))
```

I also made the variable bit more descriptive.

Reply:
I agree with you, the variable names are not very descriptive.

Your suggestion yields an array with the translated values, but we need a `Map` with entries `action -> translate('file_action_' + action)`.
```
> ['edited', 'renamed', 'created', 'deleted'].reduce((acc, i) => {acc[i] = 'file_action_' + i; return acc}, {})
{ edited: 'file_action_edited',
  renamed: 'file_action_renamed',
  created: 'file_action_created',
  deleted: 'file_action_deleted' }
>
> ['edited', 'renamed', 'created', 'deleted'].map((action) => 'file_action_' + action)
[ 'file_action_edited',
  'file_action_renamed',
  'file_action_created',
  'file_action_deleted' ]
```

Reply:
Ah right, sorry I didn't spot that. Happy to keep the `reduce` then.